### PR TITLE
Bugfix on array-input of evaluation call

### PR DESCRIPTION
**Pull request by VikingScientist**

### CODE REVIEW

**Comment by VikingScientist** [comment]
`LRSplineVolume.insert_knot()` crashes with segmentation fault and I can't tell why. This is why no trivariate tests are generated in the last commit.

**EDIT:** Nevermind. I messed up and generated the volume tests as well so now the test fail with segmentation fault on all configurations

**Comment by VikingScientist** [comment]
The following minimal python script crashes with segmentation fault. In this particular case, no MeshRectangle objects are expected to be deleted.

```python
import lrspline as lr

vol = lr.LRSplineVolume(3,3,3)
vol.insert_knot(.5,'u')
```

**Comment by VikingScientist** [comment]
The equivalent c++ implementation has no problems

``` c++
#include "LRSpline/LRSplineVolume.h"
#include "LRSpline/MeshRectangle.h"
#include <iostream>

using namespace LR;
using namespace std;

int main(int argc, char **argv) {

  LRSplineVolume vol(3,3,3,3,3,3);
  vol.insert_line(new MeshRectangle(.5,0,0,.5,1,1));
  cout << vol << endl;
}
```

**Comment by TheBB** [comment]
Rebased on master.

File: lrspline/__init__.py [hunk 1]
```diff
@@ -593,8 +622,18 @@ def corners(self):
...
+        for k in new_knots:
+            start = list(self.start())
+            end = list(self.end())
+            start[direction] = end[direction] = k
+            mr = raw.MeshRectangle(start[0], start[1], start[2], end[0], end[1], end[2], multiplicity)
+            self.insert(mr)
+
     def insert(self, mr):
-        self.w.insert_line(mr.w.copy())
+        self.w.insert_line(mr.copy())
```
Comment (TheBB) [inline lrspline/__init__.py]: This is a no-no. The `LRSplineVolume.insert` method should be usable on the Python side without having to know that its argument is some sort of wrapper object. Instead I've changed the `insert_knot` method to call into the raw objects directly.

**Pull request merged**